server/statistics/buckets/hot_bucket_cache.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buckets
+
+import (
+	"bytes"
+	"context"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/logutil"
+	"github.com/tikv/pd/pkg/rangetree"
+	"github.com/tikv/pd/server/core"
+	"go.uber.org/zap"
+)
+
+type status int
+
+const (
+	alive status = iota
+	archive
+)
+
+const (
+	// queue is the length of the channel used to send the statistics.
+	queue = 20000
+	// bucketBtreeDegree is the degree of the btree used to store the bucket.
+	bucketBtreeDegree = 10
+
+	// the range of the hot degree should be [-100, 100]
+	minHotDegree = -100
+	maxHotDegree = 100
+)
+
+// HotBucketCache is the cache of hot stats.
+type HotBucketCache struct {
+	tree            *rangetree.RangeTree       // regionId -> BucketTreeItem
+	bucketsOfRegion map[uint64]*BucketTreeItem // regionId -> BucketTreeItem
+	taskQueue       chan flowBucketsItemTask
+	ctx             context.Context
+}
+
+// bucketDebrisFactory returns the debris.
```

Comment:
How about using fragments instead of debris? BTW, can we make the comment clearer

Reply:
I think the the mean of debris is suitable for this function, the results of the function are the debirs if the key range of the item is bigger than the given key range.